db/db_impl/db_impl.cc

```diff
@@ -1131,16 +1132,22 @@ Status DBImpl::ApplyReplicationLogRecord(ReplicationLogRecord record,
       }
       case ReplicationLogRecord::kMemtableSwitch: {
         WriteContext write_context;
+        MemTableSwitchRecord mem_switch_record;
+        Slice contents_slice(record.contents);
+        DeserializeMemtableSwitchRecord(&contents_slice, &mem_switch_record);
+
         autovector<ColumnFamilyData*> cfds;
         SelectColumnFamiliesForAtomicFlush(&cfds);
-        for (const auto cfd : cfds) {
+
+        for (auto cfd: cfds) {
           cfd->Ref();
-          s = SwitchMemtable(cfd, &write_context, "");
+          s = SwitchMemtable(cfd, &write_context, mem_switch_record);
```

Comment:
Do we need to skip empty memtables here?

Reply:
oh I almost forgot that!  Thanks for the reminder, will add tests as well